src/shogun/evaluation/SigmoidCalibration.h

```diff
@@ -0,0 +1,133 @@
+
+		/** Destructor. */
+		virtual ~CSigmoidCalibration();
+
+		/** Get name. */
+		virtual const char* get_name() const
+		{
+			return "SigmoidCalibration";
+		}
+
+		/** Fit sigmoid parameters for binary labels.
+		* @param predictions The predictions outputted by the machine
+		* @param targets The true labels corresponding to the predictions
+		* @return boolean indicating whether the calibration was succesful
+		**/
+		virtual bool fit_binary(CBinaryLabels* predictions, CBinaryLabels* targets);
+
+		/** Calibrate binary predictions based on parameters learned by calling fit.
+		* @param predictions The predictions outputted by the machine
+		* @return Calibrated binary labels
+		**/
+		virtual CBinaryLabels* calibrate_binary(CBinaryLabels* predictions);
+
+		/** Fit calibration parameters for multiclass labels. Fits sigmoid 
+		* parameters for each class seperately.
+		* @param predictions The predictions outputted by the machine
+		* @param targets The true labels corresponding to the predictions
+		* @return boolean indicating whether the calibration was succesful
+		**/
+		virtual bool fit_multiclass(CMulticlassLabels* predictions, CMulticlassLabels* targets);
+
+		/** Calibrate multiclass predictions based on parameters learned by calling fit.
+		* The predictions are normalized over all classes.
+		* @param predictions The predictions outputted by the machine
+		* @return Calibrated binary labels
+		**/
+		virtual CMulticlassLabels* calibrate_multiclass(CMulticlassLabels* predictions);
+
+	 	/** Set maximum number of iterations
+	 	* @param maxiter maximum number of iterations
+	 	*/
+	 	void set_maxiter(index_t maxiter);
+
+	 	/** Get max iterations
+	 	* @return maximum number of iterations 
+	 	*/
+	 	index_t get_maxiter();
+
+	 	/** Set min step
+	 	* @param minstep min step taken in line search
+	 	*/
+	 	void set_minstep(float64_t minstep);
+
+	 	/** Get min step 
+	 	* @return minimum steps taken in line search
+	 	*/
+	 	float64_t get_minstep();
+
+	 	/** Set sigma
+	 	* @param sigma Set to a value greater than 0 to ensure that the Hessian matrix is positive semi-definite
+	 	*/
+	 	void set_sigma(float64_t sigma);
+
+	 	/** Get sigma
+	 	* @return sigma
+	 	*/
+	 	float64_t get_sigma();
+
+	 	/** Get epsilon
+	 	* @param epsilon stopping criteria
+	 	*/
+	 	void set_epsilon(float64_t epsilon);
+
+	 	/** Get epsilon
+	 	* @return stopping critera
+	 	*/
+	 	float64_t get_epsilon();
+
+	private:
+		/** Initialize parameters */
+		void init();
+
+		/** Helper function that calibrates values of given vector using the given sigmoid parameters
+		* @param values The values to be calibrated
+		* @param params The sigmoid paramters to be used for calibration
+		*/
+		SGVector<float64_t> calibrate_values(SGVector<float64_t> values, CStatistics::SigmoidParamters params);
+
+	private:
+		/** Vector to store parameter A of sigmoid for each class. In case of binary labels, only one pair of parameters are stored. */
```

Comment:
no type in the comments or doxygen, it is clear from the code

Reply:
Should it the first line be something like: "Stores parameter A of sigmoid for each class"?

Reply:
Yes, but that is really me nitpicking. Just for the future